File: agents/classifier.py
# agents/classifier.py
from langchain_core.prompts import ChatPromptTemplate
from langchain_google_genai import ChatGoogleGenerativeAI
from core.models import Task
from core.config import MODEL
import os

_GOOGLE_API_KEY = os.getenv("GOOGLE_API_KEY")

_CLASSIFIER_PROMPT = ChatPromptTemplate.from_messages([
    ("system",
     "You are a cognitive effort classifier for tasks.\n"
     "Return one of low|medium|high and a confidence 0–1.\n"  # <-- fixed: no braces
     "Guidelines:\n"
     "- Deep work (reports, coding, analysis, research, design) → high.\n"
     "- Meetings, reviews, writing short notes → medium.\n"
     "- Email, quick calls, errands, admin → low.\n"
     "Only classify based on the given title and notes."),
    ("human", "Task: {title}\nNotes: {notes}")
])

_llm = ChatGoogleGenerativeAI(
    model=MODEL,
    temperature=0,
    google_api_key=_GOOGLE_API_KEY,
)

def classify_effort(task: Task) -> Task:
    out = _llm.invoke(
        _CLASSIFIER_PROMPT.format_messages(
            title=task.title,
            notes=task.notes or ""
        )
    )
    text = (out.content or "").lower()

    effort, conf = "medium", 0.6
    if "high" in text:
        effort, conf = "high", 0.8
    elif "low" in text:
        effort, conf = "low", 0.7
    elif "medium" in text:
        effort, conf = "medium", 0.7

    # simple keyword nudges
    hi_kw = ["report", "analysis", "prototype", "research", "design", "study"]
    if any(k in (task.title + " " + (task.notes or "")).lower() for k in hi_kw):
        effort, conf = "high", max(conf, 0.85)

    lo_kw = ["email", "call", "text", "schedule", "calendar", "meeting"]
    if any(k in (task.title + " " + (task.notes or "")).lower() for k in lo_kw) and effort != "high":
        effort, conf = "low", max(conf, 0.8)

    task.effort, task.confidence = effort, conf
    return task

File: bootcamp-boiler/lib/agents/classifier.ts
// lib/agents/classifier.ts - Task effort classification

import { getModel } from './gemini';
import { Task } from '../types';

const CLASSIFIER_PROMPT = `You are a cognitive effort classifier for tasks.
Return one of low|medium|high and a confidence 0–1.
Guidelines:
- Deep work (reports, coding, analysis, research, design) → high.
- Meetings, reviews, writing short notes → medium.
- Email, quick calls, errands, admin → low.
Only classify based on the given title and notes.`;

export async function classifyEffort(task: Task): Promise<Task> {
  try {
    const model = getModel();
    const prompt = `${CLASSIFIER_PROMPT}\n\nTask: ${task.title}\nNotes: ${task.notes || ''}`;
    
    const result = await model.generateContent(prompt);
    const text = (result.response.text() || '').toLowerCase();

    let effort: Task['effort'] = 'medium';
    let confidence = 0.6;

    if (text.includes('high')) {
      effort = 'high';
      confidence = 0.8;
    } else if (text.includes('low')) {
      effort = 'low';
      confidence = 0.7;
    } else if (text.includes('medium')) {
      effort = 'medium';
      confidence = 0.7;
    }

    // Keyword-based overrides
    const taskText = (task.title + ' ' + (task.notes || '')).toLowerCase();
    const highKeywords = ['report', 'analysis', 'prototype', 'research', 'design', 'study'];
    if (highKeywords.some(k => taskText.includes(k))) {
      effort = 'high';
      confidence = Math.max(confidence, 0.85);
    }

    const lowKeywords = ['email', 'call', 'text', 'schedule', 'calendar', 'meeting'];
    if (lowKeywords.some(k => taskText.includes(k)) && effort !== 'high') {
      effort = 'low';
      confidence = Math.max(confidence, 0.8);
    }

    return {
      ...task,
      effort,
      confidence
    };
  } catch (error) {
    console.error('Classification error:', error);
    return {
      ...task,
      effort: 'medium',
      confidence: 0.5
    };
  }
}